### DISCUSSION TITLE
How to pass variable from custom node to sidebar

### DISCUSSION THREAD

**Discussion by sujit-baniya** [xyflow/xyflow#2690] [Q&A]
### Please describe the feature that you want to propose

I'm trying to use react-flow in one of the project and add a feature to open sidebar whenever a button on any custom node is clicked. The sidebar includes config of each node. Not sure how could I implement this feature.

![image](https://user-images.githubusercontent.com/4959215/207490565-3ce67390-f384-43b5-8f87-72aa8fd078c4.png)

**Comment by amintabrizi** [comment]
It's a little late but I implemented this Sandbox for your problem.
https://codesandbox.io/s/eloquent-lucy-2tudrh?file=/src/App.js
